edge-functions/serverside-analytics/pages/index.tsx

```diff
@@ -0,0 +1,164 @@
+import Head from 'next/head'
+import { Layout, Text, Page, Code, Link, Button } from '@vercel/examples-ui'
+import useSwr from 'swr'
+
+import fetchAPI from '../lib/fetchApi'
+import { FC, useState } from 'react'
+import { useRouter } from 'next/router'
+
+const Snippet: FC = ({ children }) => {
+  return (
+    <pre className="border-accents-2 border rounded-md bg-white overflow-x-auto p-6 transition-all">
+      {children}
+    </pre>
+  )
+}
+
+function Home() {
+  const { data, error } = useSwr('/views', fetchAPI, { refreshInterval: 500 })
+  const router = useRouter()
+  return (
+    <Page>
+      <Head>
+        <title>Serverside Analytics - Vercel Example</title>
+        <meta
+          name="description"
+          content="Vercel example how to use Serverside Analytics"
+        />
+        <link rel="icon" href="/favicon.ico" />
+      </Head>
+
+      <section className="flex flex-col gap-6">
+        <Text variant="h1">Serverside Analytics usage example</Text>
+        <Text>
+          Next.js Middleware provides a great way to track page views and
+          implement analytics. This example shows how to use{' '}
+          <Link href="https://nextjs.org/docs/middleware">Middleware</Link> in
+          order to count the number of page views. This example contains two
+          pages: <Code>/</Code> and <Code>/other</Code>. We are using Middleware
+          to log each visit to the pages:
+        </Text>
+        {}
+        <ul>
+          <li>
+            <Text variant="description">
+              Home page received {data?.home && data.home} views{' '}
+              {router.asPath === '/' && '(Current page)'}
+            </Text>
+          </li>
+          <li>
+            <Text variant="description">
+              Other page received {data?.other && data.other} views{' '}
+            </Text>
+          </li>
+        </ul>
+        <Text>
+          <Button onClick={router.reload}>Reload Page</Button>
+          <Button className="ml-4" onClick={() => router.push('/other')}>
+            Switch Page
+          </Button>
+        </Text>
+        <Text>
+          Our Middleware function sends the data in a{' '}
+          <Link href="https://supabase.com/" target="_blank">
+            Supabase
+          </Link>{' '}
+          table. To keep the function light and quick, we are using their rest
+          API endpoint instead of the{' '}
+          <Link href="https://github.com/supabase/supabase-js" target="_blank">
+            SDK
+          </Link>{' '}
+          to avoid the extra loading time to our function.
```

Comment:
In case is needed create a screen component to reuse instead of copy pasting the content